pkg/oc/cli/cmd/newapp.go

```diff
@@ -154,6 +156,46 @@ type NewAppOptions struct {
 	*ObjectGeneratorOptions
 }
 
+func checkResources(discoveryClient discovery.DiscoveryInterface, items []runtime.Object) error {
+	resources, err := discoveryClient.ServerResources()
+	if err != nil {
+		return fmt.Errorf("Unable to to get list of available resources: %v", err)
+	}
+
+	resourceMap := make(map[schema.GroupVersionKind]bool)
+	for _, resourceList := range resources {
+		for _, resource := range resourceList.APIResources {
+			gvk := schema.GroupVersionKind{
+				Group:   resource.Group,
+				Version: resource.Version,
+				Kind:    resource.Kind,
+			}
+			resourceMap[gvk] = true
+		}
+	}
+
+	declaredResources := make(map[schema.GroupVersionKind]bool)
+	for _, item := range items {
+		declaredResources[item.GetObjectKind().GroupVersionKind()] = true
```

Comment:
Your problem here is that you're comparing internal objects (`items` - you're getting as input with empty `GroupVersionKind`)  with external (`resources` - you're getting from discovery). The simplest approach for now is to convert the internals to externals, in your case change these two lines into:
```go
versioned := kcmdutil.AsDefaultVersionedOrOriginal(item, nil)
declaredResources[versioned.GetObjectKind().GroupVersionKind()] = true
``` 
and it will work as you imagine. 
@juanvallejo I'm guessing we'll need to figure out what to do with new-app... sigh...